awsclilinter/awsclilinter/cli.py

```diff
@@ -0,0 +1,162 @@
+    while True:
+        choice = input(prompt).lower().strip()
+        if choice in ["y", "n", "u", "q"]:
+            return choice
+        print("Invalid choice. Please enter y, n, u, or q.")
+
+
+def display_finding(finding: LintFinding, index: int, total: int, script_content: str):
+    """Display a finding to the user with context."""
+    src_lines = script_content.splitlines()
+    dest_lines = finding.edit.inserted_text.splitlines()
+    start_line = finding.line_start
+    end_line = finding.line_end
+    src_lines_removed = end_line - start_line + 1
+    new_lines_added = len(dest_lines)
+
+    # Create a map from line numbers to their indices within the full script file
+    line_positions = []
+    pos = 0
+    for i, line in enumerate(src_lines):
+        line_positions.append((pos, pos + len(line)))
+        pos += len(line) + 1
+
+    # Get context lines
+    context_start = max(0, start_line - CONTEXT_SIZE)
+    context_end = min(len(src_lines), end_line + CONTEXT_SIZE + 1)
+    src_context_size = context_end - context_start
+    dest_context_size = src_context_size + (new_lines_added - src_lines_removed)
+
+    print(f"\n[{index}/{total}] {finding.rule_name}")
+    print(f"{finding.description}")
+    print(
+        f"\n{CYAN}@@ -{context_start + 1},{src_context_size} "
+        f"+{context_start + 1},{dest_context_size} @@{RESET}"
+    )
+
+    for i in range(context_start, context_end):
+        line = src_lines[i] if i < len(src_lines) else ""
+
+        if start_line <= i <= end_line:
+            # This line is being modified
+            print(f"{RED}-{line}{RESET}")
+
+            if i == end_line:
+                line_start_pos, _ = line_positions[i]
+                start_pos_in_line = max(0, finding.edit.start_pos - line_start_pos)
+                end_pos_in_line = min(len(line), finding.edit.end_pos - line_start_pos)
+                new_line = line[:start_pos_in_line] + finding.edit.inserted_text + line[end_pos_in_line:]
+                # In case the inserted text takes up multiple lines,
+                # inject a + at the start of each line.
+                new_line = new_line.replace("\n", "\n+")
+                # Print the new line suggestion.
+                print(f"{GREEN}+{new_line}{RESET}")
+        else:
+            # Context line
+            print(f"{line}")
+
+
+def interactive_mode(findings: List[LintFinding], script_content: str) -> List[LintFinding]:
+    """Run interactive mode and return accepted findings."""
+    accepted = []
+    for i, finding in enumerate(findings, 1):
+        display_finding(finding, i, len(findings), script_content)
+        choice = get_user_choice("\nApply this fix? [y]es, [n]o, [u]pdate all, [q]uit: ")
+
+        if choice == "y":
+            accepted.append(finding)
+        elif choice == "u":
+            accepted.extend(findings[i - 1 :])
+            break
+        elif choice == "q":
+            print("Cancelled.")
+            sys.exit(0)
```

Comment:
It might be a bit confusing if a user accepts a few fixes and then presses `q` to exit early with the expectation that the accepted fixes would be applied.
- We should explicitly document this behavior.
- [non-blocking suggestion] And also consider adding a [s]ave option that "flushes" accepted fixes and applies them. If you think we should do this, okay with doing it in a followup PR.

Reply:
Hmm let's discuss this with the team. I can see advantages to different approaches here